### PR TITLE
Implement test case for policy-psp 

### PR DESCRIPTION
**Pull request by ChunxiAlexLuo**
https://github.com/stolostron/backlog/issues/20845

Signed-off-by: Chunxi Luo <chuluo@redhat.com>

### CODE REVIEW

File: test/integration/policy_psp_test.go [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+			defaultTimeoutSeconds*4,
+			1,
+		).Should(Equal(policiesv1.Compliant))
+	})
+
+	It("The PodSecurityPolicy "+pspName+" should exist", func() {
+		By("Checking the PodSecurityPolicy " + pspName)
+		Eventually(
+			func() error {
+				_, err := clientManagedDynamic.Resource(pspGVR).Get(
```
Comment (ycao56) [inline test/integration/policy_psp_test.go:120]: You can use `clentManaged` to get psp as this is not a CRD.

File: test/e2e/policySet_e2e_test.go [hunk 2]
```diff
@@ -34,6 +35,7 @@ var _ = Describe("Test policy set", func() {
 			By("Creating " + testPolicySetYaml + " result is " + output)
 			Expect(err).To(BeNil())
 
+			By("Checking " + testPolicyName + " exists on the hub in ns " + userNamespace)
```
Comment (ycao56) [inline test/e2e/policySet_e2e_test.go:38]: Please move these changes to separate commits/PRs and they are not related to PSP test case.

File: test/integration/policy_psp_test.go [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+					context.TODO(), pspName, metav1.GetOptions{},
+				)
```
Comment (dhaiducek) [inline test/integration/policy_psp_test.go:122]:
```suggestion
				_, err := clientManaged.PolicyV1beta1().PodSecurityPolicies().Get(
					context.TODO(), pspName, metav1.GetOptions{},
				)
```

**Review by dhaiducek — DISMISSED** [review]
We don't have to add a GVR for this because it's a native Kubernetes resource.

**Comment by dhaiducek** [comment]
Oops--sorry for the duplicate review comments, @ChunxiAlexLuo! I didn't realize @ycao56 had already said the same thing! 🙂

File: test/integration/policy_psp_test.go [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+	const (
+		rootPolicyName   = "policy-podsecuritypolicy"
+		rootPolicyURL    = policyCollectSCURL + "policy-psp.yaml"
+		rootPolicyNSName = "policy-psp"
+		pspName          = "sample-restricted-psp"
+	)
+
+	It("stable/"+rootPolicyName+" should be created on the hub cluster", func() {
+		By("Creating " + rootPolicyName + " on the hub")
+		output, err := utils.KubectlWithOutput(
```
Comment (ycao56) [inline test/integration/policy_psp_test.go:29]:
```suggestion
		_, err := utils.KubectlWithOutput(
```
The output is already printed in `utils.KubectlWithOutput`. So you can just ignore the return value.

File: test/integration/policy_psp_test.go [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+			"apply",
+			"-f",
+			rootPolicyURL,
+			"-n",
+			userNamespace,
+			"--kubeconfig="+kubeconfigHub,
+		)
+		By("Creating " + rootPolicyName + " result on the hub is " + output)
```
Comment (ycao56) [inline test/integration/policy_psp_test.go:37]:
```suggestion
```

**Comment by ycao56** [comment]
@ChunxiAlexLuo Please move any changes other than `policy_psp_test.go` to a separate commit or PR. This makes commit history cleaner.

**Comment by ChunxiAlexLuo** [comment]
> @ChunxiAlexLuo Please move any changes other than `policy_psp_test.go` to a separate commit or PR. This makes commit history cleaner.

The rest changes are from `fmt` auto-formatting/fixing, I will remove them

**Comment by ycao56** [comment]
Looks good! @ChunxiAlexLuo Please update the commit message and PR name and then it is good to go.
